### PR TITLE
implement CAST(col as type)

### LINKED ISSUE #398: `CAST` not supported
```limbo> select
        l_returnflag,
        l_linestatus,
        sum(l_quantity) as sum_qty,
        sum(l_extendedprice) as sum_base_price,
        avg(l_quantity) as avg_qty,
        avg(l_extendedprice) as avg_price,
        avg(l_discount) as avg_disc,
        count(*) as count_order
from
        lineitem
where
        l_shipdate <= cast('1998-12-01' as datetime)
group by
        l_returnflag,
        l_linestatus
order by
        l_returnflag,
        l_linestatus;
thread 'main' panicked at core/translate/expr.rs:719:35:
not yet implemented
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace```

### PR DESCRIPTION
**Pull request by jussisaurio**
Closes #398

### CODE REVIEW

**Comment by penberg** [comment]
@jussisaurio Please drop `testing/testing.db-wal` from the PR.

File: core/vdbe/mod.rs [hunk 1]
```diff
@@ -2667,6 +2679,183 @@ fn exec_if(reg: &OwnedValue, null_reg: &OwnedValue, not: bool) -> bool {
...
+}
+
+/// For tables not declared as STRICT, the affinity of a column is determined by the declared type of the column, according to the following rules in the order shown:
+/// If the declared type contains the string "INT" then it is assigned INTEGER affinity.
+/// If the declared type of the column contains any of the strings "CHAR", "CLOB", or "TEXT" then that column has TEXT affinity. Notice that the type VARCHAR contains the string "CHAR" and is thus assigned TEXT affinity.
+/// If the declared type for a column contains the string "BLOB" or if no type is specified then the column has affinity BLOB.
+/// If the declared type for a column contains any of the strings "REAL", "FLOA", or "DOUB" then the column has REAL affinity.
+/// Otherwise, the affinity is NUMERIC.
+/// Note that the order of the rules for determining column affinity is important. A column whose declared type is "CHARINT" will match both rules 1 and 2 but the first rule takes precedence and so the column affinity will be INTEGER.
+fn affinity(datatype: &str) -> &str {
```
Comment (pereman2) [inline core/vdbe/mod.rs:2758]: Would it be better to return an Enum so that we don't compare strings and we compare numbers instead?

**Review by pereman2 — APPROVED** [review]

File: core/vdbe/mod.rs (hunk 1 above)
Comment (jussisaurio) [reply core/vdbe/mod.rs]: yes it would

context switching between this and javascript...

Comment (jussisaurio) [reply core/vdbe/mod.rs]: fixed

**Pull request merged**